panel/widgets/chat.py

```diff
@@ -0,0 +1,1533 @@
+        attempt to use the first renderer that does not raise an
+        exception. If None, will attempt to infer the renderer
+        from the value.""")
+
+    placeholder_text = param.String(default="", doc="""
+        If placeholder is the default LoadingSpinner,
+        the text to display next to it.""")
+
+    placeholder_threshold = param.Number(default=1, bounds=(0, None), doc="""
+        Min duration in seconds of buffering before displaying the placeholder.
+        If 0, the placeholder will be disabled.""")
+
+    auto_scroll_limit = param.Integer(default=200, bounds=(0, None), doc="""
+        Max pixel distance from the latest object in the Column to
+        activate automatic scrolling upon update. Setting to 0
+        disables auto-scrolling.""")
+
+    scroll_button_threshold = param.Integer(default=100, bounds=(0, None), doc="""
+        Min pixel distance from the latest object in the Column to
+        display the scroll button. Setting to 0
+        disables the scroll button.""")
+
+    view_latest = param.Boolean(default=True, doc="""
+        Whether to scroll to the latest object on init. If not
+        enabled the view will be on the first object.""")
+    value = param.List(item_type=ChatEntry, doc="""
+        The list of entries in the feed.""")
+
+    _placeholder = param.ClassSelector(class_=ChatEntry, allow_refs=False, doc="""
+        The placeholder wrapped in a ChatEntry object;
+        primarily to prevent recursion error in _update_placeholder.""")
+
+    _disabled = param.Boolean(default=False, doc="""
+        Whether the chat feed is disabled.""")
+
+    _stylesheets: ClassVar[List[str]] = [
+        f"{CDN_DIST}css/chat_feed.css"
+    ]
+
+    _composite_type: ClassVar[Type[ListPanel]] = Card
+
+    def __init__(self, **params):
+        if params.get("renderers") and not isinstance(params["renderers"], list):
+            params["renderers"] = [params["renderers"]]
+        super().__init__(**params)
+        # instantiate the card
+        card_params = {
+            "header": self.header,
+            "hide_header": self.header is None,
+            "collapsed": False,
+            "collapsible": False,
+            "css_classes": ["chat-feed"],
+            "header_css_classes": ["chat-feed-header"],
+            "title_css_classes": ["chat-feed-title"],
+            "sizing_mode": self.sizing_mode,
+            "height": self.height,
+            "width": self.width,
+            "max_width": self.max_width,
+            "max_height": self.max_height,
+        }
+        card_params.update(**self.card_params)
+        if self.sizing_mode is None:
+            card_params["height"] = card_params.get("height", 500)
+        self._composite.param.update(**card_params)
+        self._composite.stylesheets = self._stylesheets
+        self._composite.styles = {"border": "1px solid var(--panel-border-color, #e1e1e1)", "padding": "0px"}
```

Comment:
When you're updating multiple parameters at once always do so via `.param.update`

```suggestion
            "max_height": self.max_height,
            "styles": {"border": "1px solid var(--panel-border-color, #e1e1e1)", "padding": "0px"},
            "stylesheets": self._stylesheets
        }
        card_params.update(**self.card_params)
        if self.sizing_mode is None:
            card_params["height"] = card_params.get("height", 500)
        self._composite.param.update(**card_params)
```